# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../glibc/nptl-i386.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./nptl/sysdeps/unix/sysv/linux/i386/lowlevellock.h.orig	2006-09-05 10:44:25.000000000 -0400
+++ ./nptl/sysdeps/unix/sysv/linux/i386/lowlevellock.h	2007-07-16 19:57:56.000000000 -0400
@@ -444,8 +444,7 @@
 #define LLL_LOCK_INITIALIZER_LOCKED	(1)
 
 
-extern int __lll_lock_wait (int val, int *__futex)
-     __attribute ((regparm (2))) attribute_hidden;
+extern void __lll_lock_wait (int *__futex) attribute_hidden;
 extern int __lll_unlock_wake (int *__futex)
      __attribute ((regparm (1))) attribute_hidden;
 extern int lll_unlock_wake_cb (int *__futex) attribute_hidden;
--- libc/nptl/sysdeps/i386/pthread_spin_trylock.c.jj	2004-05-09 19:42:23.000000000 +0200
+++ libc/nptl/sysdeps/i386/pthread_spin_trylock.c	2004-05-10 12:01:42.000000000 +0200
@@ -0,0 +1,46 @@
+/* Copyright (C) 2004 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+   Contributed by Jakub Jelinek <jakub@redhat.com>, 2004.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#include "pthreadP.h"
+#include <errno.h>
+
+#ifndef LOCK_PREFIX
+# ifdef UP
+#  define LOCK_PREFIX	/* nothing */
+# else
+#  define LOCK_PREFIX	"lock;"
+# endif
+#endif
+
+
+int
+pthread_spin_trylock (lock)
+     pthread_spinlock_t *lock;
+{
+  int ret = EBUSY;
+  asm ("\n"
+       LOCK_PREFIX "btrl $0, %0\n\t"
+       "jnc 1f\n\t"
+       "xorl %1, %1\n\t"
+       "1:"
+       : "=m" (*lock), "=r" (ret)
+       : "m" (*lock), "1" (ret));
+
+  return ret;
+}
